internal/backend/grpc/management.go

```diff
@@ -654,6 +656,81 @@ func (s *managementServer) ReadAuditLog(req *management.ReadAuditLogRequest, srv
 	return closeFn()
 }
 
+func (s *managementServer) RebootMachine(ctx context.Context, req *management.RebootMachineRequest) (*management.RebootMachineResponse, error) {
+	if _, err := s.authCheckGRPC(ctx, auth.WithRole(role.Admin)); err != nil {
+		return nil, err
+	}
+
+	ctx = actor.MarkContextAsInternalActor(ctx)
+
+	if _, err := safe.StateGetByID[*infra.Machine](ctx, s.omniState, req.MachineId); err != nil {
```

Comment:
Maybe a bit crazy idea, but we could make reboot work seamlessly for all kind of machines.
If the infra machine is not found we can fall back to the talos API call.
What do you think?

Reply:
Sure, we can do that. I'm just concerned with the fact that this endpoint does not block or report any status (like stream the phases back or something), so not sure how much value it actually brings.

Reply:
Does it block in case of `omnictl power reboot`?

Reply:
nope, none. It blocks until reboot is issued, not until machine comes back.

Reply:
so it will be more or less consistent I guess

Reply:
another crazy idea is to intercept `talosctl reboot` command and make it go through the infra provider 🙃